minimesos/src/main/java/com/containersol/minimesos/container/AbstractContainer.java

```diff
@@ -28,21 +28,38 @@
 
     private static Logger LOGGER = Logger.getLogger(AbstractContainer.class);
 
-    private final String randomId;
+    protected String clusterId;
 
-    private String containerId = "";
+    protected String uuid;
+
+    protected String containerId = "";
 
     private boolean removed;
 
     protected DockerClient dockerClient;
-    private String clusterId;
+
     private String ipAddress = null;
 
     protected AbstractContainer(DockerClient dockerClient) {
         this.dockerClient = dockerClient;
-        this.randomId = Integer.toUnsignedString(new SecureRandom().nextInt());
+        this.uuid = Integer.toUnsignedString(new SecureRandom().nextInt());
+    }
+
+    protected void associateContainerId() {
```

Comment:
This is an expensive method. Please, add missing ID to contractors of all containers. IDs are available in `MesosCluster(String clusterId)`